### PR TITLE
Integrate settable player collisionbox into player API

### PR DESCRIPTION
**Pull request by paramat**
Attends to https://github.com/minetest/minetest_game/issues/1411#issuecomment-317105882

### CODE REVIEW

**Comment by Ferk** [comment]
I'd say, either consider it trivial and merge it already so the collision box is set in the game and not the engine now, or commit it together with the new model that changes the size of the player so they can be adjusted at the same time.

Otherwise both tickets (this and the one changing the model) would have to synchronize.

**Comment by stujones11** [comment]
Code seems fine but it should probably get a mention in game_api.txt https://github.com/minetest/minetest_game/blob/master/game_api.txt#L677

**Comment by paramat** [comment]
No longer WIP, it seems correct to use the true height of the 3D player model for the sprite.

The new shorter model may be controversial so i think this needs to be merged before that.

File: mods/default/player.lua [hunk 1]
```diff
@@ -58,12 +59,14 @@ function default.player_set_model(player, model_name)
...
 			visual = "mesh",
 			visual_size = model.visual_size or {x=1, y=1},
+			collisionbox = model.collisionbox or {-0.3, 0.0, -0.3, 0.3, 1.77, 0.3},
 		})
 		default.player_set_animation(player, "stand")
 	else
 		player:set_properties({
 			textures = { "player.png", "player_back.png", },
 			visual = "upright_sprite",
+			collisionbox = {-0.3, 0.0, -0.3, 0.3, 1.77, 0.3},
```
Comment (stujones11) [inline mods/default/player.lua]: Maybe leave this at 1.75? I have not tested how it looks, just guessing it would've be designed that way.

Comment (paramat) [inline mods/default/player.lua]: Yeah perhaps, that was the original value after all.

Comment (paramat) [inline mods/default/player.lua]: Done.

**Comment by paramat** [comment]
Updated with suggestion.

**Comment by paramat** [comment]
I'll rebase to add to players mod when #1849 is merged.

**Comment by paramat** [comment]
Replaced by #1863

**Pull request closed without merging**